pkg/datastoresrp/setup/setup.go

```diff
@@ -0,0 +1,167 @@
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateRedisCacheTimeout,
+			AsyncOperationRetryAfter: AsyncOperationRetryAfter,
+		},
+		Delete: builder.Operation[datamodel.RedisCache]{
+			AsyncJobController: func(options asyncctrl.Options) (asyncctrl.Controller, error) {
+				return pr_ctrl.NewDeleteResource[*datamodel.RedisCache, datamodel.RedisCache](options, &rds_proc.Processor{}, recipeControllerConfig.Engine, recipeControllerConfig.ConfigLoader)
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateRedisCacheTimeout,
+			AsyncOperationRetryAfter: AsyncOperationRetryAfter,
+		},
+		Custom: map[string]builder.Operation[datamodel.RedisCache]{
+			"listsecrets": {
+				APIController: rds_ctrl.NewListSecretsRedisCache,
+			},
+		},
+	})
+
+	_ = ns.AddResource("mongoDatabases", &builder.ResourceOption[*datamodel.MongoDatabase, datamodel.MongoDatabase]{
+		RequestConverter:  converter.MongoDatabaseDataModelFromVersioned,
+		ResponseConverter: converter.MongoDatabaseDataModelToVersioned,
+
+		Put: builder.Operation[datamodel.MongoDatabase]{
+			UpdateFilters: []apictrl.UpdateFilter[datamodel.MongoDatabase]{
+				rp_frontend.PrepareRadiusResource[*datamodel.MongoDatabase],
+			},
+			AsyncJobController: func(options asyncctrl.Options) (asyncctrl.Controller, error) {
+				return pr_ctrl.NewCreateOrUpdateResource[*datamodel.MongoDatabase, datamodel.MongoDatabase](options, &mongo_proc.Processor{}, recipeControllerConfig.Engine, recipeControllerConfig.ResourceClient, recipeControllerConfig.ConfigLoader)
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateMongoDatabaseTimeout,
+			AsyncOperationRetryAfter: AsyncOperationRetryAfter,
+		},
+		Patch: builder.Operation[datamodel.MongoDatabase]{
+			UpdateFilters: []apictrl.UpdateFilter[datamodel.MongoDatabase]{
+				rp_frontend.PrepareRadiusResource[*datamodel.MongoDatabase],
+			},
+			AsyncJobController: func(options asyncctrl.Options) (asyncctrl.Controller, error) {
+				return pr_ctrl.NewCreateOrUpdateResource[*datamodel.MongoDatabase, datamodel.MongoDatabase](options, &mongo_proc.Processor{}, recipeControllerConfig.Engine, recipeControllerConfig.ResourceClient, recipeControllerConfig.ConfigLoader)
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateMongoDatabaseTimeout,
+			AsyncOperationRetryAfter: AsyncOperationRetryAfter,
+		},
+		Delete: builder.Operation[datamodel.MongoDatabase]{
+			AsyncJobController: func(options asyncctrl.Options) (asyncctrl.Controller, error) {
+				return pr_ctrl.NewDeleteResource[*datamodel.MongoDatabase, datamodel.MongoDatabase](options, &mongo_proc.Processor{}, recipeControllerConfig.Engine, recipeControllerConfig.ConfigLoader)
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateRedisCacheTimeout,
```

Comment:
We should use "AsyncDeleteMongoDatabaseTimeout"
https://github.com/project-radius/radius/blob/4bda9604f26c593cc3aa3deeb965dbc08c1c3497/pkg/datastoresrp/frontend/controller/types.go#L32